src/node.cc

```diff
@@ -840,6 +842,26 @@ void SetupNextTick(const FunctionCallbackInfo<Value>& args) {
   args.GetReturnValue().Set(ret);
 }
 
+void WeakCallback(
+    const v8::WeakCallbackInfo<v8::Persistent<v8::Promise,
+      node::ResetInDestructorPersistentTraits<v8::Promise> > >& data) {
+  Environment* env = Environment::GetCurrent(data.GetIsolate());
+  Persistent<Promise>* p = data.GetParameter();
+  Local<Promise> promise = p->Get(env->isolate());
+  Local<Value> err = promise->Result();
+  Persistent<Value>* e = new Persistent<Value>(env->isolate(), err);
```

Comment:
It's not safe to create a new persistent in a weak callback, V8 iterates over the `new_space_nodes_` list that `Persistent<T>()` mutates.  You could revive (with `ClearWeak()`) the promise.